pkg/prometheus/statefulset.go

```diff
@@ -496,3 +496,15 @@ func ShareProcessNamespace(p monitoringv1.PrometheusInterface) *bool {
 		) == monitoringv1.ProcessSignalReloadStrategyType,
 	)
 }
+
+func GetStatupProbePeriodSecondsAndFailureThreshold(cfp monitoringv1.CommonPrometheusFields) (int32, int32) {
+	var startupPeriodSeconds int32 = 15
+	var startupFailureThreshold int32 = 60
+
+	if ptr.Deref(cfp.MaximumStartupDurationSeconds, 0) > 0 {
+		startupPeriodSeconds = *cfp.MaximumStartupDurationSeconds / 60
```

Comment:
Maybe we could have a rough heuristic to determine the "correct" period (e.g. we don't want to check 1) too few if the duration is small or 2) too often if the duration is large)?
* If max duration <= 60s then period = 10s
* If max duration > 60s then period = 20s

Or we just keep the default 15s and avoid the complexity :smile:

Reply:
I'd say if the max duration < 60s we should use default other wise we're fine to use the max duration defined by the user.